src/lib.rs

```diff
@@ -22,7 +22,7 @@
 //! # {
 //! let raw_secret_key = std::fs::read("./test_assets/secret_key.asc")?;
 //! let pkg = rpm::RPMBuilder::new("test", "1.0.0", "MIT", "x86_64", "some awesome package")
-//!             .compression(rpm::Compressor::from_str("gzip")?)
+//!             .compression(rpm::CompressionType::Gzip)
```

Comment:
`CompressionDetails` currently uses `&str` for `compression_type` and `compression_level`, it should use `CompressionType` for the former and `u8` for the latter.

What we're missing here is some way for the user to specify the compression level to use.  `CompressionDetails` would actually be nice for this.

So how about we implement the signature like this: `fn compressor<T: Into<CompressionDetails>>(details: T)` and implement `Into<CompressionDetails> for CompressionType` with the default values.  And then a function implemented on `CompressionType` such as:

```
CompressionType::with_type(level: u8)
```

with some checks to make sure it's a reasonable value

`Compressor` doesn't need to be public anymore, just `pub(crate)`

Reply:
@dralley, it's a good suggestion! Well, I will try to create small POC and will post it here.

Reply:
@dralley well, I have one problem. I cannot create function with signature: `fn compressor<T: Into<CompressionDetails>>(details: T)`. Only: `fn compressor<T: TryInto<CompressionDetails>>(details: T) -> Result<(), RPMError>`. If this is ok I will push changes.

Reply:
@dralley, hello! Can you look your idea in my implementation: https://github.com/rpm-rs/rpm/pull/135